src/http_proxy_stub.c

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <limits.h>
+#include <stddef.h>
+#include "azure_c_shared_utility/gballoc.h"
+#include "azure_c_shared_utility/xio.h"
+#include "azure_c_shared_utility/socketio.h"
+#include "azure_c_shared_utility/crt_abstractions.h"
+#include "azure_c_shared_utility/http_proxy_io.h"
+#include "azure_c_shared_utility/base64.h"
+
+static CONCRETE_IO_HANDLE http_proxy_stub_create(void* io_create_parameters)
+{
+    CONCRETE_IO_HANDLE result;
+
+    (void)io_create_parameters;
+    result = NULL;
+
+    return result;
+}
+
+static void http_proxy_stub_destroy(CONCRETE_IO_HANDLE http_proxy_io)
+{
+    (void)http_proxy_io;
+
+    LogError("Function %s is a stub and should never be called", __FUNCTION__);
+}
+
+static int http_proxy_stub_open(CONCRETE_IO_HANDLE http_proxy_io, ON_IO_OPEN_COMPLETE on_io_open_complete, void* on_io_open_complete_context, ON_BYTES_RECEIVED on_bytes_received, void* on_bytes_received_context, ON_IO_ERROR on_io_error, void* on_io_error_context)
+{
+    int result;
+
+    (void)http_proxy_io;
+    (void)on_io_open_complete;
+    (void)on_io_open_complete_context;
+    (void)on_bytes_received;
+    (void)on_bytes_received_context;
+    (void)on_io_error;
+    (void)on_io_error_context;
+
+    LogError("Function %s is a stub and should never be called", __FUNCTION__);
+    result = __LINE__;
```

Comment:
>__LINE__ [](start = 13, length = 8)

Can you change to __FAILURE__ for some devices.

Reply:
In case the markdown threw us off - its `__FAILURE__`